pkg/schedule/schedulers/balance_range.go

```diff
@@ -74,9 +79,60 @@ type balanceRangeSchedulerJob struct {
 	Timeout time.Duration   `json:"timeout"`
 	Ranges  []core.KeyRange `json:"ranges"`
 	Alias   string          `json:"alias"`
+	Start   *time.Time      `json:"start,omitempty"`
+	Finish  *time.Time      `json:"finish,omitempty"`
+	Create  time.Time       `json:"create"`
 	Status  JobStatus       `json:"status"`
 }
 
+func (conf *balanceRangeSchedulerConfig) begin(job *balanceRangeSchedulerJob) bool {
+	conf.Lock()
+	defer conf.Unlock()
+	if job.Status != pending {
+		return false
+	}
+	now := time.Now()
+	job.Start = &now
+	job.Status = running
+	if err := conf.save(); err != nil {
```

Comment:
It is not recommended that
1. modify the input param
2. call conf.save() to persist the modified input struct

I prefer more explicit implement. For example, we can use `jobs map[JobID]balanceRangeSchedulerJob` and input `JobID` here rather than Job.

Reply:
@rleungx   the functions must ensure that jobs list must contains the given job.